Assets/Scripts/Army/Army.cs

```diff
@@ -7,14 +7,90 @@
 using UnityEngine.Events;
 using Mirror;
 
-[RequireComponent(typeof(ArmyMovement))]
+[RequireComponent(typeof(ArmyMovement), typeof(ArmyHealth), typeof(ArmyVisuals))]
 public class Army : Entity
 {
+    // State variables
+    [SyncVar] private ArmyState state = ArmyState.Idle;
+    public ArmyState State => state;
+
+    // Unit variables
     readonly SyncList<Unit> armyUnits = new SyncList<Unit>(); // Change to set if necessary
-    public ReadOnlyCollection<Unit> ArmyUnits => new ReadOnlyCollection<Unit>(armyUnits);
+    public SyncList<Unit> ArmyUnits => armyUnits;
+
+
+    // Attack variables
+    [Header("Attack settings")]
+    [SyncVar] private float attackDamage = 0f;
+    [SyncVar][SerializeField] private float attackRange = 5f;
+    [SyncVar] private Entity attackTarget = null; // Only used on the server
+
+    // MonoBehaviour references
+    ArmyVisuals armyVisuals;
+    ArmyHealth  armyHealth;
+
+    #region Events
+    public static event Action<Army> ServerOnArmySpawned;
+    public static event Action<Army> ServerOnArmyDespawned;
+
+    public static event Action<Army> AuthorityOnArmySpawned;
+    public static event Action<Army> AuthorityOnArmyDespawned;
     
+    #endregion
+
     public Army() { }
 
+    void Awake()
+    {
+        entityMovement = GetComponent<ArmyMovement>();
+        entityHealth = GetComponent<ArmyHealth>();
+        armyHealth = entityHealth as ArmyHealth;
+        if(armyHealth == null)
+            Debug.LogError("ArmyHealth is null"); 
+        armyVisuals = GetComponent<ArmyVisuals>();
+    }
+
+    void Update()
+    {
+        if (isServer) // Handle game logic
+        {
+            
+            if (state == ArmyState.Attacking)
+            {
+                if (attackTarget == null) // Target died
+                {
+                    Debug.Log("Target died");
+                    state = ArmyState.Idle;
+                }
+                else
+                {
+                    if (Vector3.Distance(transform.position, attackTarget.transform.position) <= attackRange)
+                    {
+                        Debug.Log("Attacking target");
+                        entityMovement.Stop();
+                        attackTarget.EntityHealth.TakeDamage(attackDamage * Time.deltaTime);
+                    }
+                    else
+                    {
+                        Debug.Log("Target out of range");
+                        entityMovement.Move(attackTarget.transform.position);
+                    }
+                }
+            }
+        }
+
+        if (isClient) // Handle client visuals
```

Comment:
This separation is good, one other way to do a ClientRpc call in the isServer part that has a `DrawDeathRay` call